src/foam/parse/ModelParserFactory.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.parse;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+import foam.core.ClassInfo;
+import foam.core.PropertyInfo;
+import foam.lib.json.Whitespace;
+import foam.lib.parse.Alt;
+import foam.lib.parse.Optional;
+import foam.lib.parse.Parser;
+import foam.lib.parse.Repeat0;
+import foam.lib.parse.Seq;
+import foam.lib.parse.Seq1;
+
+public class ModelParserFactory {

```

Comment:
I think we should call this ModelQueryParserFactory since it's building a QueryParser for a specific model, not a value parser like the ModelParserFactory in foam/lib/json does.

Also let's move this and related classes to a subpackage.